src/metabase/upload.clj

```diff
@@ -678,7 +676,54 @@
 ;;; |  hydrate based_on_upload for FE
 ;;; +--------------------------------
 
-(mi/define-simple-hydration-method based-on-upload
+(defn uploadable-table-ids
+  "Returns the subset of table ids where the user can upload to the table."
+  [table-ids]
+  (if (empty? table-ids)
```

Comment:
Does toucan / honeysql have a fast "no query" return path for `[:in []]` queries? It just seems a bit sad to hand roll this optimization 😄

Reply:
they don't